src/main/scala/io/scalac/wtf/domain/User.scala

```diff
@@ -1,11 +1,46 @@
 package io.scalac.wtf.domain
 
+import cats.data.Validated.{invalidNel, valid}
+import cats.data.{NonEmptyList, ValidatedNel}
+import cats.implicits._
 import slick.lifted.MappedTo
 
+import scala.util.matching.Regex
 
 case class UserId(value: Long) extends MappedTo[Long]
 
 case class User(
     id: Option[UserId] = None,
     email: String, 
     password: String)
+
+object User {
+
+  sealed trait ValidationError
+  final case object WrongEmailPattern extends ValidationError
+  final case object PasswordTooShort  extends ValidationError
+
+  def validateUser(email: String, password: String): ValidatedNel[ValidationError, User] = {
+    val validation = (validateEmail(email) |@| validatePassword(password))
```

Comment:
I can extend it, no problem.
Examples of error stacking/propagation in a list is shown in the tests